system.go

```diff
@@ -28,6 +28,32 @@ type SystemInput struct {
 	Note        *string          `json:"note,omitempty"`
 }
 
+func (s *System) GetTag(tagId ID, client *Client) *Tag {
+	if s == nil {
+		return nil
+	}
+	tags, err := s.SystemId.Tags(client, nil)
+	if err != nil {
+		fmt.Println(fmt.Errorf("Error getting tags: %s", err))
+		return nil
+	}
+	for _, tag := range tags.Nodes {
+		if tag.Id == tagId {
+			return &tag
+		}
+	}
```

Comment:
Since we do this on every function we should refactor this to be a utitlity function on a `TagConnection` 

so you can just do `return tags.FindTag(tagId)`